okhttp-tests/src/test/java/com/squareup/okhttp/internal/http/HeadersTest.java

```diff
@@ -145,6 +146,47 @@
     assertEquals("OkHttp", headers.value(0));
   }
 
+  @Test public void addParsing() {
+    Headers headers = new Headers.Builder()
+        .add("foo: bar")
+        .add("foo : bar")
+        .add(" foo: bar")
+        .add("ping:  pong  ")
+        .add("kit:kat")
+        .add(" : baz")
+        .build();
+    assertEquals(Arrays.asList("bar"), headers.values("foo"));
+    assertEquals(Arrays.asList("bar"), headers.values("foo "));
+    assertEquals(Arrays.asList("bar"), headers.values(" foo"));
+    assertEquals(Arrays.asList("pong"), headers.values("ping"));
+    assertEquals(Arrays.asList("kat"), headers.values("kit"));
+    assertEquals(Arrays.asList("baz"), headers.values(" "));
```

Comment:
this case is potentially sketchy. The spec is strict about what we should produce, and it says that ` ` is not a valid header name.
http://tools.ietf.org/html/rfc7230#section-3.2

I think we have to reject this input.